src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -0,0 +1,411 @@
+
+    /**
+     * Returns the maximum capacity of active entities.
+     */
+    val capacity: Int
+        get() = entityService.capacity
+
+    /**
+     * Returns the world's systems.
+     */
+    val systems: Array<IntervalSystem>
+        get() = systemService.systems
+
+    init {
+        val worldCfg = WorldConfiguration().apply(cfg)
+        componentService = ComponentService(worldCfg.componentFactory)
+        entityService = EntityService(worldCfg.entityCapacity, componentService)
+        val injectables = worldCfg.injectables
+        // add the world as a used dependency in case any system or ComponentListener needs it
+        injectables["World"] = Injectable(this, true)
+
+        // set a Fleks internal global reference to the current world that
+        // gets created. This is used to correctly initialize the world
+        // reference of any created system and to correctly register
+        // any created FamilyListener below.
+        CURRENT_WORLD = this
+
+        Inject.injectObjects = injectables
+        Inject.mapperObjects = componentService.mappers
+
+        // create and register ComponentListener
+        // it is important to do this BEFORE creating systems because if a system's init block
+        // is creating entities then ComponentListener already need to be registered to get notified
+        worldCfg.compListenerFactory.forEach {
+            val compType = it.key
+            val listener = it.value.invoke()
+            val mapper = componentService.mapper(compType)
+            mapper.addComponentListenerInternal(listener)
+        }
+
+        // create and register FamilyListener
+        // like ComponentListener this must happen before systems are created
+        worldCfg.famListenerFactory.forEach {
+            val (listenerType, factory) = it
+            try {
+                val listener = factory.invoke()
+                FamilyListener.CURRENT_FAMILY.addFamilyListener(listener)
+            } catch (e: Exception) {
+                if (e is FleksFamilyException) {
+                    throw FleksFamilyListenerCreationException(
+                        listenerType,
+                        "FamilyListener must define at least one of AllOf, NoneOf or AnyOf"
+                    )
+                }
+                throw e
+            }
+        }
+
+        // create systems
+        systemService = SystemService(worldCfg.systemFactory)
+
+        // verify that there are no unused injectables
+        val unusedInjectables = injectables.filterValues { !it.used }.map { it.value.injObj::class }
+        if (unusedInjectables.isNotEmpty()) {
+            throw FleksUnusedInjectablesException(unusedInjectables)
+        }
+
+        // clear dependencies at the end because they are no longer necessary,
+        // and we don't want to keep a reference to them
+        Inject.injectObjects = EMPTY_INJECTIONS
+        Inject.mapperObjects = EMPTY_MAPPERS
```

Comment:
I found my problem. I was accessing the injectObjects property after creation of my specific systems and component listeners. This was not intended. Thus I will change my code. No changes on your side needed.